crates/solidity/outputs/cargo/crate/src/extensions/semver/mod.rs

```diff
@@ -0,0 +1,463 @@
+pub mod parser;
+pub mod tests;
+
+use std::fmt::Display;
+
+use semver::Version;
+
+use crate::cst::NonterminalKind;
+use crate::utils::LanguageFacts;
+
+/// Parse the version pragmas in the given Solidity source code and return a list of language
+/// versions that can fulfill those requirements.
+pub fn infer_language_version(src: &str) -> Vec<Version> {
+    let parser = crate::parser::Parser::create(LanguageFacts::LATEST_VERSION).unwrap();
+    let output = parser.parse_file_contents(src);
+
+    let mut cursor = output.create_tree_cursor();
+
+    let mut found_ranges = Vec::<Range>::new();
+    while cursor.go_to_next_nonterminal_with_kind(NonterminalKind::VersionPragma) {
```

Comment:
I suggest iterating on `VersionExpressionSets` instead, as this will remove the need to strip `solidity` below.